draft-ietf-tls-tls13.md

```diff
@@ -4319,6 +4322,31 @@ will receive various TLS 1.x versions in this field, however its value
 MUST always be ignored.
 
 
+## Zero-RTT backwards compatibility
+
+A TLS 1.3 ClientHello with 0-RTT data is not compatible with older servers
+implementing previous versions of TLS. This can cause compatibility issues if a
+client first connects to a 0-RTT-capable TLS 1.3 server and later to an older
+server, both behind the same service. For example, a multi-server deployment
+may deploy TLS 1.3 gradually with some servers implementing TLS 1.3 and some
+implementing TLS 1.2. This may also occur if a TLS 1.3 deployment gets reverted
+to TLS 1.2.
+
+If a TLS 1.3 client sends a ClientHello with 0-RTT to an older server, the
+server will respond with a ServerHello containing an older version number.
+(The handshake will later fail at the 0-RTT data.) Clients which accept older
+versions of TLS SHOULD then retry the connection advertising TLS 1.3 without
+offering 0-RTT.
```

Comment:
Please change this "SHOULD" to a "MAY", as I think that's more appropriate. If a client sends a hello to a server it knows supports 0RTT, yet it borks on it, it should not always be expected to retry as this could easily be considered evidence that it's not connecting to the server it thinks it is. (e.g. a controlled environment where a known set of clients are interoperating with a known set of servers) If the client wants to fail here, that's up to the implementers. Some will generally want to retry; others could be fine with failing. Going into more detail would be outside the scope of this document.